demo: ctrl-c crashes demo with `--no-line-editor`

```
./cockroach demo --no-line-editor 

CTRL-C

demo@127.0.0.1:26257/movr> ^C
attempting to cancel query...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x429c844]

goroutine 5111 [running]:
github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).maybeHandleInterrupt.func2()
        github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2366 +0x1c4
created by github.com/cockroachdb/cockroach/pkg/cli/clisqlshell.(*cliState).maybeHandleInterrupt
        github.com/cockroachdb/cockroach/pkg/cli/clisqlshell/sql.go:2344 +0x118
```


Jira issue: CRDB-20871

--- a/pkg/cli/clisqlshell/sql.go
+++ b/pkg/cli/clisqlshell/sql.go
@@ -2358,6 +2358,7 @@ func (c *cliState) maybeHandleInterrupt() func() {
 					// by re-throwing the signal after stopping the signal capture.
 					signal.Reset(os.Interrupt)
 					_ = sysutil.InterruptSelf()
+					return
 				}
 
 				fmt.Fprintf(c.iCtx.stderr, "\nattempting to cancel query...\n")

--- a/pkg/cli/interactive_tests/test_interrupt.tcl
+++ b/pkg/cli/interactive_tests/test_interrupt.tcl
@@ -1,7 +1,5 @@
 #! /usr/bin/env expect -f
 
-# Disabled until https://github.com/cockroachdb/cockroach/issues/76483 is resolved.
-
 source [file join [file dirname $argv0] common.tcl]
 
 start_server $argv
@@ -47,9 +45,32 @@ send "\r"
 eexpect "defaultdb>"
 end_test
 
-# Quit the SQL client, and open a unix shell.
+# Quit the SQL client.
 send_eof
 eexpect eof
+
+start_test "Check that interrupt without a line editor quits the shell."
+# regression test for #90653.
+spawn $argv sql --no-line-editor
+eexpect "defaultdb>"
+interrupt
+set timeout 1
+expect {
+    "attempting to cancel query" { exit 1 }
+    "panic: runtime error" { exit 1 }
+    timeout {}
+}
+set timeout 30
+interrupt
+expect {
+    "attempting to cancel query" { exit 1 }
+    "panic: runtime error" { exit 1 }
+    eof {}
+}
+end_test
+
+
+# Open a unix shell.
 spawn /bin/bash
 set shell2_spawn_id $spawn_id
 send "PS1=':''/# '\r"

